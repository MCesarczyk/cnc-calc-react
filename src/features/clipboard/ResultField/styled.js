import styled, { css } from "styled-components";

export const ResultWrapper = styled.div`
  display: flex;
  flex-basis: 16rem;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-basis: unset;
    width: 100%;
  }
`

export const Result = styled.div`
    min-width: 16rem;
    border: solid 1px ${({ theme }) => theme.color.primary};
    border-radius: 5px;
    padding: 5px;
    color: ${({ theme }) => theme.color.primaryText};

    ${({ placeholder }) => placeholder && css`
      color: ${({ theme }) => theme.color.placeholder};
    `}
`;

export const ResultCopyButton = styled.button`
  background: transparent;
  border: none;
  outline: none;
  color: ${({ theme }) => theme.color.copyButtonFont};
  margin-left: 0.5rem;

  &:hover {
    filter: brightness(1.5);
  }

  &:active {
    filter: brightness(0.5);
  }
`