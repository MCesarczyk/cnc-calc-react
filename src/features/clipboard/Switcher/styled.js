import styled from "styled-components";

export const SwitcherWrapper = styled.div`
  margin-top: 1.5rem;
  padding-left: 2.5rem;
`

export const SwitcherLabel = styled.label`
  font-variant: small-caps;
  font-weight: 700;
  margin-right: 1rem;
`

export const Switcher = styled.button``

export const SwitcherIndicator = styled.div``