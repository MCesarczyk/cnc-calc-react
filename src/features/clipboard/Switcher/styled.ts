import styled, { css } from 'styled-components';

export const SwitcherWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 1.5rem;
`;

export const SwitcherLabel = styled.label`
  font-variant: small-caps;
  font-weight: 700;
  margin-right: 1rem;
  cursor: pointer;
`;

export const Switcher = styled.button`
  width: 3rem;
  height: 2rem;
  background-color: #d9d9d9;
  border: 2px solid #525252;
  border-radius: 1rem;
`;

export const SwitcherIndicator = styled.div<{ $memorymode?: boolean }>`
  width: 2rem;
  height: 2rem;
  border: 2px solid #525252;
  border-radius: 1rem;
  background-color: #d9d9d9;
  position: relative;
  left: -0.5rem;
  top: -3px;
  transition: all 0.3s ease-in-out;

  ${({ $memorymode }) =>
    $memorymode &&
    css`
      transform: translateX(1.2rem);
      background-color: #1e3246;
    `}
`;
