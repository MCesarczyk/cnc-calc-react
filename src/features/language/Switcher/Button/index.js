import { useContext } from "react";
import LanguageContext from "../../context";
import { changeLanguage, isButtonActive } from "../functions";
import { SwitcherButton } from "./styled";

const Button = ({ language }) => {
  const { langId, setLangId } = useContext(LanguageContext);

  const onLanguageChange = (key) => setLangId(changeLanguage(key));

  return (
    <SwitcherButton
      onClick={() => onLanguageChange(language.key)}
      style={{ backgroundImage: `url(${language.flag})` }}
      active={isButtonActive(language.key, langId)}
    >
      {language.key}
    </SwitcherButton>
  )
};

export default Button;