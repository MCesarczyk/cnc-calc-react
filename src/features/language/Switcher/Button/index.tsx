import { useContext } from 'react';

import LanguageContext from '../../context';
import { changeLanguage, isButtonActive } from '../functions';
import { SwitcherButton } from './styled';
import { Language } from '../../types';

interface ButtonProps {
  language: {
    key: Language;
    flag: string;
  };
}

const Button = ({ language }: ButtonProps) => {
  const { langId, setLangId } = useContext(LanguageContext);

  const onLanguageChange = (key: Language) => setLangId(changeLanguage(key) as Language);

  return (
    <SwitcherButton
      onClick={() => onLanguageChange(language.key)}
      style={{ backgroundImage: `url(${language.flag})` }}
      $active={isButtonActive(language.key, langId) ? 'active' : undefined}
    >
      {language.key}
    </SwitcherButton>
  );
};

export default Button;
