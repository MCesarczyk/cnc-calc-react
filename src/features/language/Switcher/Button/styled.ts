import styled, { css } from 'styled-components';

export const SwitcherButton = styled.button<{ $active?: string }>`
  color: ${({ theme }) => theme.color.primary};
  background-color: ${({ theme }) => theme.color.primaryContrast};
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  border: 2px solid ${({ theme }) => theme.color.primary};
  border-radius: 2px;

  ${({ $active }) =>
    $active === 'active' &&
    css`
      border-color: ${({ theme }) => theme.color.primaryContrast};
    `}

  &:hover {
    filter: brightness(125%);
  }

  &:active {
    filter: brightness(150%);
  }
`;
