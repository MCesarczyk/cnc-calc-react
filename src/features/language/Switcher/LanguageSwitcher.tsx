import styled from 'styled-components';

import languages from 'features/language/languages';
import { SwitcherButton } from './SwitcherButton';
import { Language } from '../types';

export const LanguageSwitcher = () => (
  <SwitcherWrapper>
    {Object.values(languages).map((language) => (
      <SwitcherButton key={language.key} language={language as { key: Language; flag: string }} />
    ))}
  </SwitcherWrapper>
);

export const SwitcherWrapper = styled.div`
  display: flex;

  @media (max-width: ${({ theme }) => theme.breakpoint.sm}) {
    flex-direction: column;
  }
`;
