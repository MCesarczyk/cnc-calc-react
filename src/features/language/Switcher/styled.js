import styled from "styled-components";

export const SwitcherWrapper = styled.div`
    display: flex;
    
    @media (max-width: ${({ theme }) => theme.breakpoint.sm}) {
        flex-direction: column;
    }
`;