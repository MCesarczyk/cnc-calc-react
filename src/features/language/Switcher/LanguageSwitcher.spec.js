import { fireEvent, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../theme';
import LanguageContext from '../context';
import languages from '../../../assets/fixtures/languages';
import LanguageSwitcher from '.';
import { SwitcherButton } from './Button/styled';
import Button from './Button';

let langId = 0;
const setLangId = (id) => (langId = id);

const switcherSetup = () =>
  render(
    <ThemeProvider theme={theme}>
      <LanguageContext.Provider value={{ langId, setLangId }}>
        <LanguageSwitcher />
      </LanguageContext.Provider>
    </ThemeProvider>,
  );

const buttonSetup = (currentLangId) => {
  const language = languages[currentLangId];

  return render(
    <ThemeProvider theme={theme}>
      <LanguageContext.Provider value={{ langId, setLangId }}>
        <Button language={language} />
      </LanguageContext.Provider>
    </ThemeProvider>,
  );
};

test('Language switcher should render button for every language', () => {
  switcherSetup();

  const buttons = screen.getAllByRole('button');
  expect(buttons.length).toEqual(languages.length);
});

test('Button sets proper language on click', () => {
  switcherSetup();

  expect(langId).toBe(0);
  const button_EN = screen.getByText('EN');
  const button_PL = screen.getByText('PL');
  fireEvent.click(button_PL);
  expect(langId).toBe(1);
  fireEvent.click(button_PL);
  expect(langId).toBe(1);
  fireEvent.click(button_EN);
  expect(langId).toBe(0);
  fireEvent.click(button_EN);
  expect(langId).toBe(0);
});

test.skip('Active button should have light border', () => {
  const activeButton = buttonSetup(0);

  const activeButtonItem = activeButton.getByRole('button');
  expect(activeButtonItem).toHaveStyle({ 'border-color': '#fff' });
});

test.skip('Inactive button should have dark border', () => {
  const inactiveButton = buttonSetup(1);

  const inactiveButtonItem = inactiveButton.getByRole('button');
  expect(inactiveButtonItem).toHaveStyle({ 'border-color': '#1e3246' });
});

test('Basic button should render properly', () => {
  const buttonItem = renderer.create(
    <ThemeProvider theme={theme}>
      <SwitcherButton />
    </ThemeProvider>,
  );

  let buttonView = buttonItem.toJSON();
  expect(buttonView).toMatchSnapshot();
});
