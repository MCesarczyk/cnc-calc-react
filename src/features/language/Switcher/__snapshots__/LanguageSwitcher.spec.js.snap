// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic button should render properly 1`] = `
.c0 {
  color: #1e3246;
  background-color: #fff;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  border: 2px solid #1e3246;
  border-radius: 2px;
}

.c0:hover {
  -webkit-filter: brightness(125%);
  filter: brightness(125%);
}

.c0:active {
  -webkit-filter: brightness(150%);
  filter: brightness(150%);
}

<button
  className="c0"
/>
`;
